gaps/gap-25_golem_certificates/gap-25_golem_certificates.md

```diff
@@ -30,46 +31,68 @@ When designing certificates structure that will allow to meet requirements menti
+If this extension is present, it must be marked critical.
 
+When processing this extension, the verifying party will check if the certificate contains all permissions required to fulfil the requested operation. It may ignore all other permission values (including ones it cannot process) as those have no effect on executing the request.
 
-| Name                       | OID                      | Semantic                                                             |
-| -------------------------- | ------------------------ | -------------------------------------------------------------------- |
-| GolemFactory               | 1.3.6.1.4.1.60000 (TODO) | Root OID for Golem Factory                                           |
-| GolemCertificates          | ${GolemFactory}.1        | Root for Golem certificates parameters                               |
-| GolemCertificatesAllowance | ${GolemCertificates}.1   | List of permissions to sign certain Golem features using certificate |
+In a chain of certificates a certificate is only valid if the Golem Factory certificate extension contains a subset of the permissions of the signing certificate. The self signed root certificate is always considered to have all permissions even if this extension is not present. This can be limited by settings on the Agent application when setting up trust for the root certificate.
 
-Example values of `GolemCertificatesAllowance` field (this list will be used in further examples, to illustrate how certification works):
+Structure of the Golem Factory certificate extension:
+```
+id-gf OBJECT IDENTIFIER ::= { iso(1) identified-organization(3) dod(6) internet(1) private(4) iana(1) 59850 }
+id-gf-certificate OBJECT IDENTIFIER ::= { id-gf 1 }
+id-gf-certificate-extension OBJECT IDENTIFIER ::= { id-gf-certificate 1 }
 
 
-| Field             | Semantic                                                                        |
-| ----------------- | ------------------------------------------------------------------------------- |
-| all               | Certificate is allowed to sign any feature (even not existing future use cases) |
-| manifest-outbound | Allowed to sign manifests with outbound network enabled                         |
-| inbound           | Allowed to sign inbound allowance for Requestors.                               |
-| ...               | {List is non-exhaustive - new features will be created in future}               |
+GolemFactoryCertificateExtension ::= CHOICE {
+       permitAll BOOLEAN,
+       permissions SEQUENCE SIZE (1..MAX) OF GolemPermissionId
 
-### Permissions chains
+GolemPermissionID ::= OBJECT IDENTIFIER
```

Comment:
it would be worth thinking about some permission attributes, e.g. to be able to say that a given certificate allows you to connect to given addresses.